public/js/user/donateDom.js

```diff
@@ -1,42 +1,133 @@
-const personalSection = document.querySelector('.personal');
-const personalForm = document.querySelector('.personal--form');
-const next = document.querySelector('.next');
-const donatSection = document.querySelector('.donation');
-const donationForm = document.querySelector('.donation--form');
-const done = document.querySelector('.submit');
-const popUpSection = document.querySelector('.popUpBack');
-const back = document.querySelector('.back');
-const firstTap = document.querySelector('.taps--peronalInfo');
-const secundtTap = document.querySelector('.taps--donationInfo');
+const {
+  personalSection,
+  personalForm,
+  next,
+  donatSection,
+  donationForm,
+  done,
+  popUpSection,
+  back,
+  firstTap,
+  secundtTap,
+  valditMsg,
+  validMsgDonation,
+} = querySelectors(['personalSection',
+  'personalForm',
+  'next',
+  'donatSection',
+  'donationForm',
+  'done',
+  'popUpSection',
+  'back',
+  'firstTap',
+  'secundtTap',
+  'valditMsg',
+  'validMsgDonation',
+], ['.personal',
+  '.personal--form',
+  '.next',
+  '.donation',
+  '.donation--form',
+  '.submit',
+  '.popUpBack',
+  '.back',
+  '.taps--peronalInfo',
+  '.taps--donationInfo',
+  '.validMsg',
+  '.validMsgDonation',
+]);
 
 next.addEventListener('click', (e) => {
   e.preventDefault();
-  personalSection.classList.add('hide');
-  donatSection.classList.remove('hide');
-  firstTap.classList.replace('taps--peronalInfo', 'taps--donationInfo');
-  secundtTap.classList.replace('taps--donationInfo', 'taps--peronalInfo');
+  const formData = new FormData(personalForm);
+  const personalInformation = {};
+  formData.forEach((value, key) => {
+    personalInformation[key] = value;
+  });
+  const {
+    fullName,
+    phoneNumber,
+    cityName,
+    email,
+  } = personalInformation;
+  if (!fullName) {
+    valditMsg.textContent = 'Please enter your full name';
+  } else if (!phoneNumber) {
+    valditMsg.textContent = 'Please enter your phone number';
+  } else if ((!/^[0-9]{10}$/.test(phoneNumber))) {
```

Comment:
@AbdallahAmmar96 Hey I had a comment in another PR about phone number regexes that is valid here too:
https://github.com/FACG6/enhance-a-smile/pull/94#discussion_r267339537

Reply:
ok I'll consider it